### PR TITLE
fix: renamed context to extensionContext

### PR DESCRIPTION
**Pull request by jeffb-sfdc**
### What does this PR do?
This PR renames instances of "context" to "extensionContext" to avoid confusion between ExtensionContext and WorkspaceContext variables.

### What issues does this PR fix or reference?
#@W-11171762@

### Functionality Before / Functionality After
No functionality change

### CODE REVIEW

File: packages/salesforcedx-vscode-core/src/index.ts
Comment (jeffb-sfdc) [inline packages/salesforcedx-vscode-core/src/index.ts:708]: This is the one spot where it wasn't clear what the right thing to do is.  `api` is returned by `activate()` and I'm not familiar with how this is used, so I kept it as being returned as, "workspaceContext".  This compiles and all tests pass, and I performed a light smoke test and haven't found any issues.

File: packages/salesforcedx-vscode-apex/src/index.ts
Comment (randi274) [inline packages/salesforcedx-vscode-apex/src/index.ts:142]: @jeffb-sfdc was this comment intentional?

File: packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceSourceRetrieveManifest.test.ts
Comment (randi274) [inline packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceSourceRetrieveManifest.test.ts:15]: @jeffb-sfdc was NLS meant to be removed here too?

File: packages/salesforcedx-vscode-core/test/vscode-integration/telemetry/MockExtensionContext.ts
Comment (randi274) [inline packages/salesforcedx-vscode-core/test/vscode-integration/telemetry/MockExtensionContext.ts:83]: I actually do really like this change here - much clearer what/why we're mocking 👍

File: packages/salesforcedx-vscode-apex/src/index.ts
Comment (jeffb-sfdc) [reply packages/salesforcedx-vscode-apex/src/index.ts:142]: It's not used.  I'll go and remove it.

File: packages/salesforcedx-vscode-lwc/test/vscode-integration/activation/activationMode.test.ts
Comment (randi274) [inline packages/salesforcedx-vscode-lwc/test/vscode-integration/activation/activationMode.test.ts:7]: Just checking that the directory was renamed here as well? This was the case for all of the MockContext's?

**Review by randi274 — APPROVED** [review]
Those are all my questions - I think refactoring here makes it much easier to understand the different type of contexts used, especially as folks onboard to the product. Perhaps it was clearer to previous maintainers what the differences were, but I'm supportive of changes that make we think will make it easier to develop going forward! Given that builds are working and tests are passing, I'm ok approving once @klewis-sfdc has reviewed them with you as well. Maybe we can plan for a larger spot check during next week's release too?

**Review by klewis-sfdc — COMMENTED** [review]
Thanks for the PR, Jeff!  Overall, I really like the changes from 
* 👍🏽 "context" --> "extensionContext"  and
* 👍🏽 "MockContext" --> "MockExtensionContext",  

but I'm not as in agreement with the changes to 
* ⚠️ add "Instance" into var names.

"Instance" is a technical implementation detail that IMO does not belong in the var name.  I actually prefer the expression on the left side here:
image
I would prefer not to establish or continue this naming convention.   Let me know what you think and how strongly you feel about wanting that change.  Happy to discuss more in standup today!

File: packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceSourceRetrieveManifest.test.ts
Comment (jeffb-sfdc) [reply packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceSourceRetrieveManifest.test.ts:15]: nls isn't used, so I removed it since workspaceContext (also not used) was removed.

File: packages/salesforcedx-vscode-lwc/test/vscode-integration/activation/activationMode.test.ts
Comment (jeffb-sfdc) [reply packages/salesforcedx-vscode-lwc/test/vscode-integration/activation/activationMode.test.ts:7]: No, just the file was renamed.  The path is `packages/salesforcedx-vscode-lwc/test/vscode-integration/activation/MockExtensionContext.ts`, and the parent directory, `activation` was not changed, just the filename.

**Comment by jeffb-sfdc** [comment]
> but I'm not as in agreement with the changes to
> 
> * ⚠️ add "Instance" into var names.
> 

done

**Pull request merged**
